fs/osfs.py

```diff
@@ -420,17 +420,18 @@ def _check_copy(self, src_path, dst_path, overwrite=False):
 
     if sys.version_info[:2] < (3, 8) and sendfile is not None:
 
-        _sendfile_error_codes = frozenset(
-            {
-                errno.EIO,
-                errno.EINVAL,
-                errno.ENOSYS,
-                errno.ENOTSUP,  # type: ignore
-                errno.EBADF,
-                errno.ENOTSOCK,
-                errno.EOPNOTSUPP,
-            }
-        )
+        _sendfile_error_codes = {
+            errno.EIO,
+            errno.EINVAL,
+            errno.ENOSYS,
+            errno.EBADF,
+            errno.ENOTSOCK,
+            errno.EOPNOTSUPP,
+        }  # type: MutableSet[int]
+
+        # PyPy doesn't define ENOTSUP so we have to add it conditionally.
+        if hasattr(errno, "ENOTSUP"):
+            _sendfile_error_codes.add(errno.ENOTSUP)
```

Comment:
I guess you _could_ convert `_sendfile_error_codes` back into a `frozenset` after adding (or not adding) the `errno.ENOTSUP` ?

Reply:
Does it need to be one?

Reply:
Dunno - but I guess it doesn't hurt? I'll leave it up to @willmcgugan to decide :slightly_smiling_face:

Reply:
A regular set will do the job. Don't see any scope for breakage.